### PR TITLE
Add DHT_send_packet

### PR DESCRIPTION
**Pull request by endoffile78**

### CODE REVIEW

File: toxcore/DHT.c [hunk 1]
```diff
@@ -2912,3 +2912,11 @@ int DHT_non_lan_connected(const DHT *dht)
 
     return 0;
 }
+
+int DHT_send_packet(const DHT *dht, IP_Port ip_port, const uint8_t *packet, uint16_t length)
```
Comment (GrayHatter) [inline toxcore/DHT.c]: should be static

File: toxcore/DHT.h [hunk 2]
```diff
@@ -428,5 +428,7 @@ int DHT_non_lan_connected(const DHT *dht);
 
 int addto_lists(DHT *dht, IP_Port ip_port, const uint8_t *public_key);
 
-#endif
+/* Sends a DHT packet */
+int DHT_send_packet(const DHT *dht, IP_Port ip_port, const uint8_t *packet, uint16_t length);
```
Comment (GrayHatter) [inline toxcore/DHT.h]: drop this, we don't want to expose this to the internal API.

**Comment by iphydf** [comment]
What's the plan here?

**Comment by GrayHatter** [comment]
@iphydf I actually don't remember what my plan here was. I do remember having a reason I wanted this change but for the life of me I can't remember it right now.

either way, I did ask @endoffile78 to write another function that would take all the given data and correctly format a dht_packet (one that we could test to the spec. But either he got busy, or is waiting for this to be merged before he starts on step two. Either way, until I can remember why I wanted this change. I'm going to close this pull request. @endoffile78 if you're watching, sorry mate I must have done/said something stupid... Either way step 2 is still needed (writing a function that'll make DHT packets (I have something similar in query.c #200)) if you still have the time to contribute, we'd certainly love and appreciate the help!!

**Pull request closed without merging**